app/src/main/java/com/github/sdp/ratemyepfl/activity/classrooms/ClassroomsListActivity.kt

```diff
@@ -18,54 +15,40 @@ import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
 
-
 @AndroidEntryPoint
-class ClassroomsListActivity : AppCompatActivity() {
+class ClassroomsListActivity : ReviewableListActivity<Classroom>() {
 
-    private lateinit var roomsAdapter: ReviewableAdapter
-    private lateinit var recyclerView: RecyclerView
     private val viewModel: ClassroomsListViewModel by viewModels()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_reviewable_list)
-
-        roomsAdapter = ReviewableAdapter { room -> displayRoomReviews(room as Classroom) }
-        recyclerView = findViewById(R.id.reviewableRecyclerView)
-        recyclerView.adapter = roomsAdapter
-
-        recyclerView.addItemDecoration(
-            DividerItemDecoration(applicationContext, DividerItemDecoration.VERTICAL)
-        )
 
         viewModel.getRooms().observe(this) {
             it?.let {
-                roomsAdapter.setData(it as MutableList<Reviewable>)
+                reviewableAdapter.setData(it as MutableList<Reviewable>)
             }
         }
-
     }
 
-    /* Opens RoomReviewsActivity when RecyclerView item is clicked. */
-    private fun displayRoomReviews(room: Classroom) {
+    override fun displayReviews(t: Classroom) {
```

Comment:
You can put this function in the super class I think, just pass one other argument. In this case it would be "RoomReviewsListActivity.EXTRA_CLASSROOMS_JSON".

Reply:
The problem is that I have to pass these arguments during the initialization of the `ReviewableAdapter` in `onCreate`....

Reply:
I followed your advice, I found a solution

Reply:
Good job